auth_saml/models/res_users.py

```diff
@@ -0,0 +1,219 @@
+
+        # TODO use a real token validation from LASSO
+        # TODO push into the validation result a real UPN
+        return {'user_id': login.assertion.subject.nameId.content}
+
+    @api.multi
+    def _auth_saml_signin(self, provider, validation, saml_response):
+        """ retrieve and sign into openerp the user corresponding to provider
+        and validated access token
+
+            :param provider: saml provider id (int)
+            :param validation: result of validation of access token (dict)
+            :param saml_response: saml parameters response from the IDP
+            :return: user login (str)
+            :raise: openerp.exceptions.AccessDenied if signin failed
+
+            This method can be overridden to add alternative signin methods.
+        """
+        token_osv = self.env['auth_saml.token']
+        saml_uid = validation['user_id']
+
+        user_ids = self.search(
+            [
+                ("saml_uid", "=", saml_uid),
+                ('saml_provider_id', '=', provider),
+            ]
+        )
+
+        if not user_ids:
+            raise openerp.exceptions.AccessDenied()
+
+        # TODO replace assert by proper raise... asserts do not execute in
+        # production code...
+        assert len(user_ids) == 1
+        user = user_ids[0]
+
+        # now find if a token for this user/provider already exists
+        token_ids = token_osv.search(
+            [
+                ('saml_provider_id', '=', provider),
+                ('user_id', '=', user.id),
+            ]
+        )
+
+        if token_ids:
+            token_ids.write(
+                {'saml_access_token': saml_response},
+            )
+        else:
+            token_osv.create(
+                {
+                    'saml_access_token': saml_response,
+                    'saml_provider_id': provider,
+                    'user_id': user.id,
+                },
+            )
+
+        return user.login
+
+    @api.model
+    def auth_saml(self, provider, saml_response):
+
+        validation = self._auth_saml_validate(provider, saml_response)
+
+        # required check
+        if not validation.get('user_id'):
+            raise openerp.exceptions.AccessDenied()
+
+        # retrieve and sign in user
+        login = self._auth_saml_signin(provider, validation, saml_response)
+
+        if not login:
+            raise openerp.exceptions.AccessDenied()
+
+        # return user credentials
+        return self.env.cr.dbname, login, saml_response
+
+    # This method is using the old v7 API because it is called BEFORE the login
+    def check_credentials(self, cr, uid, token):
+        """token can be a password if the user has used the normal form...
+        but we are more interested in the case when they are tokens
+        and the interesting code is inside the except clause
+        """
+        token_osv = self.pool.get('auth_saml.token')
+
+        try:
+            baseuser.check_credentials(
```

Comment:
here you bypass the auth_crypt module, so we can't connect if ``password_crypt`` is set but not password. So admin user can only connect to odoo with a fresh ``password`` set from sql or set saml_uid from sql or xml file.

I've done a test using super(..).credentials(...) while starting to migrate on odoo9, I haven't get any issues, what was wrong with super?